src/backend/cuda/blas.cpp

```diff
@@ -312,16 +340,26 @@ void gemm(Array<T> &out, af_mat_prop optLhs, af_mat_prop optRhs, const T *alpha,
 
     if (oDims.ndims() <= 2) {
         if (rDims[bColDim] == 1) {
-            dim_t incr = (optRhs == AF_MAT_NONE) ? rStrides[0] : rStrides[1];
-            if (is_same<T, half>::value) {
-                AF_ERROR(
-                    "GEMV does not support half, Please create an issue on "
-                    "the GitHub repo.",
-                    AF_ERR_NOT_SUPPORTED);
-            }
-            CUBLAS_CHECK(gemv_func<T>()(blasHandle(), lOpts, lDims[0], lDims[1],
-                                        alpha, lhs.get(), lStrides[1],
-                                        rhs.get(), incr, beta, out.get(), 1));
+            dim4 rhsTileDims((bColDim == 0 ? lDims[aRowDim] : 1),
+                             (bColDim == 1 ? lDims[aRowDim] : 1));
+
+            auto tRhs = tile(rhs, rhsTileDims);
+
+            auto t1 = (lOpts == CUBLAS_OP_C ?
+                       transpose(lhs, true) :
+                       (lOpts == CUBLAS_OP_T ? transpose(lhs, false) : lhs));
+            auto t2 = (rOpts == CUBLAS_OP_C ?
+                       conj<T>(tRhs) :
+                       (rOpts == CUBLAS_OP_T ? tRhs : transpose(tRhs, false)));
+
+            auto product = arithOp<T, af_mul_t>(t1, t2, t1.dims());
+
+            out = reduce<af_add_t, T, T>(product, 1, false, 0);
```

Comment:
The out array for this function is already allocated and the results from the reduced need to be copied into it. For this to work we will need to modify reduce which is not a great idea. For now lets allocate a new array and copy into it.